### ISSUE TITLE
Consider removing envisage.ui.single_project

### ISSUE THREAD

**Issue by rahulporuri** [enthought/envisage#327]
[`envisage.ui.single_project`](https://github.com/enthought/envisage/tree/master/envisage/ui/single_project) is outdated, unused and probably broken. Instead of continuing to work on it (flake8-ing it, updating copyright headers), it might just be better to completely remove it for the upcoming major release 5.0.0.

See related https://github.com/enthought/envisage/issues/269

**Comment by mdickinson** [comment]
+1. Let's kill it. We've made various minor fixes to it  over the past couple of years. I think that's actually bad: it gives the impression that it's maintained, usable code. But it has no tests, it has bits of dead code, and prior to some of the fixes, there were large parts of it that couldn't ever have been run, since they contained `NameError`s. It should just die.

**Comment by kitchoi** [comment]
This was referenced from #269: https://github.com/enthought/encad/issues/42
(I don't know how much it takes to just keep `envisage.ui.single_project.api.Project` around. It might not be much, or it could be a lot.)

**Comment by kitchoi** [comment]
Honestly, the issue with having wasted time on flake8 the package was mainly me not having looked into all the errors first before trying to fix some of them. If we add a comment in `setup.cfg` to deter any such future fruitless endeavour, that would also be alright.

**Comment by mdickinson** [comment]
> This was referenced from #269: [enthought/encad#42](https://github.com/enthought/encad/issues/42)

Yep. I've looked into this before, and I don't see this as a blocker for removing single-project. Anyone who wants to use encad with the latest Envisage will need to figure out what to do about it, but encad (at least, the parts of it using single-project) is also itself fairly dead. 

We've wasted too much time on that directory in the codebase already; we shouldn't waste any more.
